### PR TITLE
Gollum-lib 5.0

### PR DESCRIPTION
**Pull request by dometto**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,7 +91,13 @@ Note that `base_path` just modifies the links.
 Get the latest version of the given human or canonical page name:
 
 ```ruby
-page = wiki.page('page-name')
+page = wiki.page('/page name') # Finds pages in the root directory of the wiki that are named 'page name' with a valid extension. 
+# => <Gollum::Page>
+
+page = wiki.page('page name') # For convenience, you can leave out the '/' in front. Paths are assumed to be relative to '/'.
```
Comment (bartkamphorst) [inline README.md]: > Paths are assumed to be relative to '/'.

Is this still correct?

File: Rakefile [hunk 2]
```diff
@@ -118,14 +118,14 @@ end
 
 desc 'Create a release build'
 task :release => :build do
-  unless `git branch` =~ /^\* master$/
-    puts "You must be on the master branch to release!"
+  unless `git branch` =~ /gollum-lib-5.x/
```
Comment (bartkamphorst) [inline Rakefile]: This should be changed I guess?

File: Rakefile [hunk 3]
```diff
@@ -118,14 +118,14 @@ end
...
+    puts "You must be on the gollum-lib-5.x branch to release!"
     exit!
   end
   sh "git commit --allow-empty -a -m 'Release #{version}'"
-  sh "git pull --rebase origin master"
```
Comment (bartkamphorst) [inline Rakefile]: Same here, this should still be `master` right?

File: Rakefile [hunk 4]
```diff
@@ -118,14 +118,14 @@ end
...
+  sh "git pull --rebase origin gollum-lib-5.x"
   sh "git tag v#{version}"
-  sh "git push origin master"
+  sh "git push origin gollum-lib-5.x"
```
Comment (bartkamphorst) [inline Rakefile]: Idem.

File: gemspec.rb [hunk 5]
```diff
@@ -3,12 +3,13 @@ def specification(version, default_adapter, platform = nil)
     s.specification_version = 2 if s.respond_to? :specification_version=
     s.required_rubygems_version = Gem::Requirement.new('>= 0') if s.respond_to? :required_rubygems_version=
     s.rubygems_version = '0.0.1'
-    s.required_ruby_version = '>= 1.9'
+    s.required_ruby_version = '>= 2.1'
```
Comment (bartkamphorst) [inline gemspec.rb]: Is this still correct, or should it be '>= 2.4'?

File: gemspec.rb [hunk 6]
```diff
@@ -3,12 +3,13 @@ def specification(version, default_adapter, platform = nil)
...
 
     s.name              = 'gollum-lib'
     s.version           = version
     s.platform          = platform if platform
-    s.date              = '2018-10-01'
+    s.date              = '2020-03-12'
```
Comment (bartkamphorst) [inline gemspec.rb]: To update.

File: gemspec.rb [hunk 7]
```diff
@@ -75,27 +76,30 @@ def specification(version, default_adapter, platform = nil)
...
     lib/gollum-lib/filter/toc.rb
-    lib/gollum-lib/filter/wsd.rb
+    lib/gollum-lib/filter/yaml.rb
     lib/gollum-lib/git_access.rb
-    lib/gollum-lib/gitcode.rb
     lib/gollum-lib/helpers.rb
     lib/gollum-lib/hook.rb
     lib/gollum-lib/macro.rb
     lib/gollum-lib/macro/all_pages.rb
+    lib/gollum-lib/macro/audio.rb
```
Comment (bartkamphorst) [inline gemspec.rb]: I don't see `warn.rb`, `note.rb` or `octicon.rb`, does that make sense?

File: lib/gollum-lib/macro/global_toc.rb [hunk 8]
```diff
@@ -3,7 +3,8 @@ class Macro
     class GlobalTOC < Gollum::Macro
       def render(title = "Global Table of Contents")
         if @wiki.pages.size > 0
-          result = '<ul>' + @wiki.pages.map { |p| "<li><a href=\"/#{p.url_path}\">#{p.url_path_display}</a></li>" }.join + '</ul>'
+          prepath=@wiki.base_path.sub(/\/$/, '')
```
Comment (bartkamphorst) [inline lib/gollum-lib/macro/global_toc.rb]: Missing spaces around `=`.

File: gemspec.rb (hunk 5 above)
Comment (dometto) [reply gemspec.rb]: I think `2.4` :+1:

File: gemspec.rb (hunk 6 above)
Comment (dometto) [reply gemspec.rb]: The `rake build` task will take care of this 👍

File: README.md (hunk 1 above)
Comment (dometto) [reply README.md]: Yes -- `wiki.page('Bilbo')` looks up `/Bilbo`

File: gemspec.rb (hunk 7 above)
Comment (dometto) [reply gemspec.rb]: The file list in the gemspec will be updated automagically by the `rake build` command (or more precisely, by its `rake gemspec` subtask).

**Comment by dometto** [comment]
@bartkamphorst added a fix for https://github.com/gollum/gollum/issues/1503 in 0e43e98 -- not really best practice to add the commit here, but it's a very small change. Also resolved your other comments, thanks for the finds!

**Comment by bartkamphorst** [comment]
Cool! 👍

**Pull request merged**